docs/branching.md

```diff
@@ -410,6 +410,89 @@ to non-branching), it may be tricky to identify valid versus invalid usage.
  
     - validate results manually if in question
 
+## Merging Branches
```

Comment:
Should we also replace this with `Concatenating branches` and in the rest of the section accordingly?